src/draw/DefaultDrawer.js

```diff
@@ -61,479 +90,455 @@ class DefaultDrawer {
+   * @param componentId - The id to search for
+   * @return {null|Component} - The component or null if not found
    */
-  drawComponents(components, parentId, links, linksDefinition) {
-    const d3Container = this.d3.select(parentId === this.rootId ? `#${parentId}` : `#${parentId} .component-container`)
-      .selectAll(`.${parentId}.component`)
-      .data(components, (data) => data.id);
 
-    const d3Elements = d3Container
-      .enter().append('svg');
+  findComponentById(rootNode,componentId) {
+    let res = null
+    if (rootNode) {
+      if (rootNode.id === componentId) res = rootNode
 
-    d3Elements
-      .attr('id', (data) => data.id)
-      .attr('class', (data) => `${parentId} ${data.definition.type} component component-${data.definition.model} ${data.definition.isContainer ? 'isContainer' : ''}`)
-      .html((data) => this.resources.models[data.definition.model]);
-
-    this.__drawModels(components, parentId);
-
-    components.forEach((component) => {
-      this.setSelectionAction(component, components, links);
-      if (component.children.length > 0) {
-        this.drawComponents(component.children, component.id, links, linksDefinition);
+      if (!res && rootNode.children) {
+        res = rootNode.children.map(c => this.findComponentById(c, componentId)).filter(c=>c)[0] ?? null
       }
-    });
 
-    this.initializeComponents(d3Elements, components, parentId);
-    if (parentId === this.rootId) {
-      this.setComponentAction(components, links, linksDefinition);
-      this.setViewPortAction(this.d3.select(`#${this.rootId}`));
-    }
-
-    this.d3.select(parentId === this.rootId ? `#${parentId}` : `#${parentId} .component-container`)
-      .selectAll(`.${parentId}.component`)
-      .attr('x', (data) => data.drawOption.x)
-      .attr('y', (data) => data.drawOption.y);
 
-    d3Container.exit().remove();
-
-    this.initializeActionMenu();
-    return true;
+    }
+    return res
   }
 
   /**
-   * Get the position of the anchors attached to the link.
-   * @param link - the link we want to set the source and target anchors positions.
-   * @returns {{source: number[], target: number[]}}
+   * Estimate how tall a component will be
+   * @param item
+   * @return {Number}
+   * @private
    */
-  getAnchorPosition(link) {
-    const { sourceAnchor, targetAnchor } = this.getDefaultAnchor(link);
-
-    const root = document.querySelector(`#${this.rootId}`)
-      .getBoundingClientRect();
-    const { x: sourceX, y: sourceY } = document.querySelector(`#${link.source} > .template > [anchor=${sourceAnchor}]`)
-      .getBoundingClientRect();
-    const { x: targetX, y: targetY } = document.querySelector(`#${link.target} > .template > [anchor=${targetAnchor}]`)
-      .getBoundingClientRect();
-
-    return {
-      source: [sourceX - root.left, sourceY - root.top],
-      target: [targetX - root.left, targetY - root.top],
-    };
+
+  __getVerticalValue(item) {
```

Comment:
value of what ?

Reply:
Renamed into __getVerticalCoefficient, its purpose is to figure out how tall a component will be based on the layout.